snakemake/jobs.py

```
@@ -1189,19 +1189,17 @@ async def postprocess(
-                    "Error recording metadata for finished job "
-                    "({}). Please ensure write permissions for the "
-                    "directory {}".format(e, self.dag.workflow.persistence.path)
-                )
+        try:
+            await self.dag.workflow.persistence.finished(self)
+        except IOError as e:
+            raise WorkflowError(
+                "Error recording metadata for finished job "
+                "({}). Please ensure write permissions for the "
+                "directory {}".format(e, self.dag.workflow.persistence.path)
+            )
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)